src/mongocxx/include/mongocxx/v1/client.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/v1/client-fwd.hpp>
+
+//
+
+#include <mongocxx/v1/detail/prelude.hpp>
+
+namespace mongocxx {
+namespace v1 {
+
+///
+/// A connection to a MongoDB deployment.
+///
+/// @important This interface does NOT fully conform to the CMAP specification!
+///
+/// @see
+/// - [Connection Monitoring and Pooling (MongoDB Specifications)](https://specifications.readthedocs.io/en/latest/connection-monitoring-and-pooling/connection-monitoring-and-pooling/)
+/// - [Connection Strings (MongoDB Manual)](https://www.mongodb.com/docs/manual/reference/connection-string/)
```

Comment:
This PR proposes deferring as much documentation as possible into "See Also" links to MongoDB Manual or MongoDB Specification pages (+ occasionally mongoc API docs). C++ Driver API documentation is and should be specific to behaviors specific to the C++ Driver, not the MongoDB server behavior or other libraries.

Also added the "Important" note to several API docs clarifying that these are not CMAP-compliant, which can be relevant for certain APM event-handling, (unstructured) logging, and error handling behavior.